### ISSUE TITLE
Panic on trying to convert struct to map

### ISSUE THREAD

**Issue by dowglaz** [gookit/goutil#192]
**System (please complete the following information):**

 - OS: `linux` [e.g. linux, macOS]
 - GO Version: `1.13` [e.g. `1.13`]
 - Pkg Version: `1.1.1` [e.g. `1.1.1`]

**Describe the bug**

I'm trying to convert a struct by using `structs.TryToMap` and it panics with the following message:

> "message":"[PANIC RECOVER] reflect: call of reflect.Value.CanInterface on zero Value goroutine 130 [running]:\ngitlab.com/arivo-software-development/arivecho/v4/srv.recoverMiddleware.Recover.RecoverWithConfig.func1.1.1()\n\t/go/pkg/mod/github.com/labstack/echo/v4@v4.12.0/middleware/recover.go:99 +0x134\npanic({0xe71340?, 0x4000026420?})\n\t/usr/local/go/src/runtime/panic.go:785 +0x124\nreflect.Value.CanInterface(...)\n\t/usr/local/go/src/reflect/value.go:1468\ngithub.com/gookit/goutil/structs.structToMap({0x1015040?, 0x4000324000?, 0x40002f4b38?}, 0x4000588200, 0x40002f4b38?)\n\t/go/pkg/mod/github.com/gookit/goutil@v0.6.15/structs/convert.go:166 +0x41c\ngithub.com/gookit/goutil/structs.StructToMap({0xdda440, 0x4000324000}, {0x0, 0x0, 0x4000142700?})\n\t/go/pkg/mod/github.com/gookit/goutil@v0.6.15/structs/convert.go:117 +0x19c\ngithub.com/gookit/goutil/structs.TryToMap(...)\n\t/go/pkg/mod/github.com/gookit/goutil@v0.6.15/structs/convert.go:29"

From the logs I got, the struct has the values:
> &{ID:ABC123 RefId:1 PhotoURL: Year:1990 SomeDate:2020-01-03 15:02:15 Trim: Type: BodyType: Counter:9001 Used:false Price:20 Value:0 Amount:0 MaxAmount:0 StockNo: ExpiresAt:<nil> UpdatedBy:<nil> Location: IsFavorite:false Structure:<nil>}

**To Reproduce**

```go
// go code
type Structure struct {
	// I don't want to expose too many infos, but here it goes only a set of float64 numbers, without any complex type or poitners
}

type MyStruct struct {
	ID                string  `json:"id" gorm:"column:id"`
	RefId             int     `json:"ref_id" gorm:"column:ref_id"`
	PhotoURL          string  `json:"photoURL" gorm:"column:photo_url"`
	Year              int     `json:"year" gorm:"column:year"`
	SomeDate          string  `json:"someDate" gorm:"column:some_date"`
	Trim              string  `json:"trim" gorm:"column:trim"`
	Type              string  `json:"type" gorm:"column:type"`
	BodyType          string  `json:"bodyType" gorm:"column:body_type"`
	Counter           int     `json:"counter" gorm:"column:counter"`
	Used              bool    `json:"used" gorm:"column:used"`
	Price             float64 `json:"price" gorm:"column:price"`
	Value             float64 `json:"value" gorm:"column:value"`
	Amount            float64 `json:"amount" gorm:"-:all"`
	MaxAmount         float64 `json:"maxAmount" gorm:"column:max_amount"`
	StockNo           string  `json:"stockNo" gorm:"column:stock_no"`
	ExpiresAt         *string `json:"expiresAt" gorm:"column:expires_at"`
	UpdatedBy         *string `json:"updatedBy" gorm:"column:updated_by"`
	Location          string  `json:"location" gorm:"column:location"`
	IsFavorite        bool    `json:"is_favorite,omitempty" gorm:"-"`
	Structure         *Structure `json:"structure,omitempty" gorm:"-:all"`
}


data := MyStruct{
  ID: "ABC123",
  RefId: 1,
  Year: 1990,
  SomeDate: "2020-01-03 15:02:15",
  Counter: 9001,
  Price: 20,
}

structs.TryToMap(data) // <- it panics!
```

**Expected behavior**

It should convert successfully to `map[string]any` without panicking.

**Screenshots**

![Screenshot 2024-09-26 at 12 39 14](https://github.com/user-attachments/assets/2c2d8c05-435a-4d49-bcc2-dcd12e2efe71)

**Additional context**

I was able to "fix it" (at least it seemed to have fixed but I don't understand the implications of it, so I'm just sharing it here, but not proposing as a MR) by doing this simple change on `goutil/structs/convert.go:166`:
```go
  var zeroValue reflect.Value // <- added this line
  if field != zeroValue && field != reflect.Zero(reflect.TypeOf(field)) { // <- and this one
    if field.CanInterface() {
      mp[name] = field.Interface()
    } else if field.CanAddr() { // for unexported field
      mp[name] = reflects.UnexportedValue(field)
    }
  } // and, of course this one :P
```

**Comment by dowglaz** [comment]
Thank you for fixing it! Do you have an idea of when it would be added to a release?

**Comment by inhere** [comment]
> Thank you for fixing it so quickly! Do you have an idea of when it would be added to a release?

Hi, released the v0.6.17
